xtensa-lx-rt/src/exception/asm.rs

```diff
@@ -113,7 +113,7 @@ global_asm!(
 #[no_mangle]
 #[link_section = ".rwtext"]
 unsafe extern "C" fn save_context() {
-    cfg_asm!(
+    cfg_naked_asm!(
```

Comment:
just a random thought but I always felt using `global_asm!` like we do for RISC-V is a better thing to do than using naked-functions

Reply:
Tbh this whole thing needs a rewrite, I'll settle for working right now. I think when we look at addressing https://github.com/esp-rs/esp-hal/issues/2390 we can think about doing the same for xtensa

Reply:
_sigh_ I think we might have to do this, or bump our MSRV to a compiler that cointains `naked_asm`

Reply:
Please don't bump msrv 🥺

Reply:
Working on it, I can run all tests on an S3 and they pass, but if I try and run an example LLVM segfaults - so that's fun

Reply:
Seems to be related to LTO, will investigate a bit futher